src/pipeline.ml

```diff
@@ -88,8 +91,13 @@ let install_compiler_df ~distro ~arch ~switch ?windows_port opam_image =
   run "opam install -y %s" depext @@
   maybe_install_secondary_compiler run os_family switch @@
   entrypoint_exec (Option.to_list personality @ opam_exec) @@
-  (match os_family with `Linux | `Cygwin -> cmd "bash" | `Windows -> cmd_exec ["cmd.exe"]) @@
-  copy ~src:["Dockerfile"] ~dst:"/Dockerfile.ocaml" ()
+  (match os_family with
+   | `Linux | `Cygwin ->
+     cmd "bash" @@
+     copy ~link:true ~src:["Dockerfile"] ~dst:"/Dockerfile.ocaml" ()
```

Comment:
Does `~link` only work on Linux?

Reply:
Yes. I'll remove `` `Cygwin``.

Reply:
It's not about symlinks, it's about BuildKit `COPY --link` operation, optimizing the case where the source files don't depend on a previous build stage. This is only supported on Docker for Linux.